vmware-tanzu/nsx-operator
# Unify security policy and network policy rule name

This patch is to:
1. Unify the rule displayName for both security policy and network policy.
2. Seperate the rule with namedport build logic from normal rule build logic.
For the rule without namedport, merging multiple ports into one service entry.
So, it's not needed to expand rules to multiple NSX rules.

Test Done:
1. Create a security policy without namedport in both non-VPC mode and VPC mode.
2. Create a security policy with namedport in both non-VPC mode and VPC mode.
3. Create a security policy with namedport and user defined rule name in both non-VPC mode and VPC mode.
4. Create a network policy without namedport in VPC mode.
5. Create a network policy with namedport in VPC mode.
6. GC in both non-VPC mode and VPC mode.
7. Create a security policy with v4.1.2 branch, and then restart NSX operator with patch change
to see if operator upgrade from v4.1.2 to current patch version is ok.
